sms-sending/javascript/index.html

```diff
@@ -0,0 +1,142 @@
+  let redirectUri = 'http://localhost:8080/';
+  let clientId = 'babbc081-0761-4f16-8f56-071aa402ebcb';
+  let queueId = '4d3af653-b60c-42f1-b373-7cf2c99610fc';
+  let dropOffMessage = 'Message sent upon dropping off the call.';
+  let userPresenceTopic = 'v2.routing.queues.'+ queueId +'.conversations.calls';
+  let message, CustomerInfo, webSocket, customerAddress, conversationId,
+  communicationId, callState, customerSmsAttr, usingMobilePhone, 
+  callConversationId, channelId, connectionUri;
+
+  // jQuery call this function when DOM's ready
+  $(document).ready(function() {
+    client.loginImplicitGrant(clientId, redirectUri)
+      .then(() => {
+        console.log('Loged in');
+
+      // Create notification channel
+      return notificationsApi.postNotificationsChannels();
+      })
+      .then((data) => {
+        channelId = data.id;
+        connectionUri = data.connectUri;        
+        
+        // Subscribe to a topic
+        let body = [{
+          'id': userPresenceTopic
+        }]; 
+        notificationsApi.putNotificationsChannelSubscriptions(channelId, body);
+        
+        // Create a new webSocket object
+        webSocket = new WebSocket(connectionUri);
+        document.getElementById('stateVal').innerHTML = 'No incoming call at the moment.';  
+
+        // If a message is recieve notificationMesssage function will be called      
+        webSocket.onmessage = notificationMessage;
+    })
+    .catch((err) => console.log(err));
+  })
+
+  // Call this function when the webSocket recieves a message
+  function notificationMessage(message) { 
+    notification = JSON.parse(message.data);		
+    console.log(notification);
+    
+    // Check notification topicname
+    if (notification.topicName.toLowerCase() === userPresenceTopic.toLowerCase()) {  
+      CustomerInfo = notification.eventBody.participants.find(participant => participant.purpose === 'customer');       
+      callConversationId = notification.eventBody.id;     
+      callState = notification.eventBody.participants[0].state;    
+      usingMobilePhone = CustomerInfo.attributes.usingMobilePhone;
+      participantId = CustomerInfo.id;
+      customerAddress = CustomerInfo.address;
+      customerAddress = customerAddress.slice(4, customerAddress.length);
+      
+      document.getElementById('stateVal').innerHTML = callState + ' to ' + customerAddress;	
+
+      // Check if messageSent attribute has been added
+      if(('messageSent' in CustomerInfo.attributes) === true) {
+        customerSmsAttr = true;
+      }
+      else {
+        customerSmsAttr = false;
+      }
+      
+      // Check if all attributes are align
+      if (callState === 'disconnected' && usingMobilePhone === 'true' && customerSmsAttr === false){
+        // Call sms sending function
+        sendSms();
+      }
+    }
+  }
+
+  // Call this function to send SMS
+  function sendSms(){
+    let body = {
+      'queueId': queueId,
+      'toAddress': '+13175763352',
```

Comment:
Is this a real phone number? Should probably leave it blank or put an improperly formatted number here so it won't work unless it's replaced by a valid number.

Reply:
oh i used it to debug. as there was issue on sending SMS numbers outside US last week.